### PR TITLE
Remove extra stat calls in coordinator for Iceberg write

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

**Review by vincentpoon — APPROVED** [review]
lgtm

**Review by findepi — APPROVED** [review]

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java [hunk 1]
```diff
@@ -616,10 +616,9 @@ public ConnectorInsertTableHandle beginInsert(ConnectorSession session, Connecto
 
         AppendFiles appendFiles = transaction.newFastAppend();
         for (CommitTaskData task : commitTasks) {
-            HdfsContext context = new HdfsContext(session);
-
             DataFiles.Builder builder = DataFiles.builder(icebergTable.spec())
-                    .withInputFile(new HdfsInputFile(new Path(task.getPath()), hdfsEnvironment, context))
+                    .withPath(task.getPath())
```
Comment (phd3) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java:620]: we can do the same in `refreshMaterializedView`

**Review by phd3 — APPROVED** [review]
Looks good. Thanks!

Also, tangential, but I suppose ParquetFileWriter can be improved further with better getMetrics. but that's still distributed so less critical than this one.

File: lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriter.java [hunk 2]
```diff
@@ -100,7 +100,7 @@ public ParquetWriter(
 
     public long getWrittenBytes()
     {
-        return outputStream.size();
```
Comment (phd3) [inline lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriter.java:103]: should this go into a separate commit?

**Review by phd3 — APPROVED** [review]
